### DISCUSSION TITLE
Copy & paste from Google Docs

### DISCUSSION THREAD

**Discussion by jipiboily** [tinacms/tinacms#2726] [General]
Hey folks,

Our content marketing guy is copying & pasting his articles weekly from Google Docs. GDocs is very convenient for various things like commenting / collaborating, Grammarly working well with it, etc.

We've just setup a proof of concept with Tina + Next, to replace our Forestry + Gatsby setup.

It looks really, really promising. I'm beyond excited 🤣

But copy & pasting doesn't work so well. It's mostly ok for things like bold and links, but a bunch of extra spaces are added.

Is it something you're looking into fixing?

Or is there a way to swap the editor (which seems to be Slate, which itself is in beta?) for us? Or would it require us to run our own forks of some parts or all of Tina?

Thanks!

**Comment by jeffsee55** [comment]
> Or is there a way to swap the editor (which seems to be Slate, which itself is in beta?) for us? Or would it require us to run our own forks of some parts or all of Tina?

I would say your best bet currently is to stick with the default editor. There are some steps to serialize/deserialize this content when it's pasted to/from other rich-text editors that perhaps we can improve. The extra lines one is likely something we can address, what else are you seeing that's not working?

**Reply by jipiboily** [reply]
Sounds good! It also sounds safer and more future-proof.

Asking my marketing manager to test with a post or two and see if there are other issues he's seeing at a glance.

**Reply by L-A** [reply]
Here's a short example I can see. The rich text string:

_**Bold italic bold italic** italic italic italic italic_

... can get deserialized this way in the editor:

`***Bold italic bold italic ****italic italic italic italic *`

This looks like an end-of-tag issue, where xml-style markup tolerates a space right before the closing tag (`<bold>content </bold>`), but simply swapping it with the MD/MDX equivalent results in a floating mark (`**content **`).

If that's helpful, I can work with the aforementioned marketing manager and build an example text that exhibits the issues we can find.

**Reply by jipiboily** [reply]
L-A is working with us as a designer & front-end dev. He's leading the proof of concept with us :)

And yeah, that indeed seems to be the only other issue we can see at first glance.

**Reply by jeffsee55** [reply]
Ok sounds good! I'll turn this into an issue. Thanks

**Reply by jipiboily** [reply]
Sounds good!

**Comment by jeffsee55** [comment]
[See issue](https://github.com/tinacms/tinacms/issues/2727)

**Comment by jipiboily** [comment]
I just realized I said "extra spaces", but I meant "extra lines".

Plus the other bold and italic mix.

It's very close to be working for us, though! :)
